File: csharp/orms/UserDashboard/Models/Message.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using System;

namespace UserDashboard.Models
{
    [Table("messages")]
    public class Message
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MinLength(5),MaxLength(255)]
        public string Content { get; set; }
        public DateTime CreatedAt { get; set; }
        List<Comment> HasComments { get; set; }
        
        [ForeignKey("Id")]
        public int UserId { get; set; }
        public User Creator { get; set; }

        public Message()
        {
            CreatedAt = DateTime.Now;
        }
    }
}

File: mean/mongodb/message-board/server/models/message.js
const mongoose = require("mongoose");

const CommentSchema = new mongoose.Schema({
    name: {type: String, required: [true, "Your name is required."], minlength: 2, maxlength: 45}, 
    content: {type: String, required: [true, "A comment is required."], minlength: 2, maxlength: 255}
    },
    {timestamps: true}
);

mongoose.model("Comment", CommentSchema);

const MessageSchema = new mongoose.Schema({
    name: {type: String, required: [true, "Your name is required."], minlength: 2, maxlength: 45}, 
    content: {type: String, required: [true, "A message is required."], minlength: 5, maxlength: 255},
    comments: {type: [CommentSchema]}
    },
    {timestamps: true}
);

mongoose.model("Message", MessageSchema);